src/invision/components/movie-list/movie-list.js

```diff
@@ -1,90 +1,133 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import thumbnail from './thumbnail.png';
 import MovieItem from '../movie-item/movie-item'
 import './movie-list.scss';
 import MovieService from '../../services/movie.service';
 
-export default class MovieList extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      movieList: [],
-      sortValue: 'RELEASE DATE',
-    };
+import MoviePopup from '../movie-popup/movie.popup';
+import DeletePopup from '../delete-movie/delete-popup';
+import useFetchMovie from './fetch-movie.hook';
 
-    this.getMovieList();
+export default function MovieList() {
 
-    this.sort = this.sort.bind(this);
-  }
+  const [movieState, updateMovieState] = useState({
+    movieList: [],
+    sortValue: 'RELEASE DATE',
+    isEditMovieVisible: false,
+    isDeleteMovieVisible: false,
+    selectedMovie: null
+  })
 
-  getMovieList() {
+  useFetchMovie(() => {
     new MovieService().getMovieList().then(movieList => {
-      this.setState({ movieList });
+      updateMovieState({ ...movieState, movieList });
     });
-  }
+  })
 
-  sort(ev) {
-    this.setState({ sortValue: ev.target.value });
+  function sort(sortValue) {
     let sortType = 'genre';
-    let sortFn = this.stringComparison;
-    if (ev.target.value === 'RELEASE DATE') {
+    let sortFn = stringComparison;
+    if (sortValue === 'RELEASE DATE') {
       sortType = 'releasedDate';
-      sortFn = this.numberComparison;
+      sortFn = numberComparison;
     }
-    const sortedMovieList = this.state.movieList.sort((movie1, movie2) => {
+    const sortedMovieList = movieState.movieList.sort((movie1, movie2) => {
       return sortFn(movie1[sortType], movie2[sortType]);
     });
 
-    this.setState({ movieList: sortedMovieList });
+    updateMovieState({ ...movieState, movieList: sortedMovieList, sortValue });
+
+    function stringComparison(b, a) {
+      return a.localeCompare(b);
+    }
+
+    function numberComparison(b, a) {
+      if (a < b)
+        return -1;
+      if (a > b)
+        return 1;
+      return 0;
+    }
 
   }
-  stringComparison(b, a) {
-    return a.localeCompare(b);
+
+  const onModalClose = ({ visible, updatedMovie }) => {
+    if (updatedMovie) {
+      const index = movieState.movieList.findIndex(movie => movie.id === updatedMovie.id);
+      movieState.movieList[index] = { ...updatedMovie };
+    }
+    updateMovieState({ ...movieState, isEditMovieVisible: visible })
   }
 
-  numberComparison(b, a) {
-    if (a < b)
-      return -1;
-    if (a > b)
-      return 1;
-    return 0;
+  const onDeletePopupClose = (isDelete) => {
+    if (isDelete) {
+      // delete move code
+    } else {
+      updateMovieState({ ...movieState, isDeleteMovieVisible: false });
+    }
   }
 
+  const findMovie = useCallback((id) => {
```

Comment:
It is not proper situation when you should use **useCallback**. You use useCallback to minimaze rerender process of the child elements. [documentation](https://ru.reactjs.org/docs/hooks-reference.html#usecallback), [additional material](https://dmitripavlutin.com/dont-overuse-react-usecallback/)

Reply:
I am a bit confused. which one should I use here

Reply:
There you should use a normal function. If you need to prevent rerender of a child component, wrap function in useCallback and use this new function as a property of the child component. Please, read articles, you can find real examples there.